lib/pre_assembly/digital_object.rb

```diff
@@ -277,8 +277,8 @@ def register_in_dor(params)
           all_pids=sourceid_pids << @pid
           all_pids.each do |pid|
             begin
+              Dor::SearchService.solr.delete_by_id(pid)  # should be unnecessary, but handles an edge case where the object is not in Fedora, but is in Solr
```

Comment:
is there ever a situation where the opposite is true?  would it be better to put the calls in independent `begin/rescue` blocks?

that's a totally naive question since i haven't really worked w/ pre-assembly, but i've run into both situations in argo, so seemed worth asking.